### PR TITLE
Add MEND and allow BA to update feed source

### PR DESCRIPTION
**Pull request by robmen**
Fixes wixtoolset/issues#5568

### CODE REVIEW

File: src/WixToolset.Mba.Core/IBootstrapperEngine.cs [hunk 1]
```diff
@@ -495,6 +500,11 @@ public enum RequestState
         /// </summary>
         Present,
 
+        /// <summary>
+        /// 
+        /// </summary>
+        Mend,
```
Comment (rseanhall) [inline src/WixToolset.Mba.Core/IBootstrapperEngine.cs:506]: If you don't mind, I would like real documentation here since this is a new concept.

**Review by rseanhall — COMMENTED** [review]
You're going to have to update to the latest Burn headers.

File: src/balutil/inc/IBootstrapperEngine.h [hunk 2]
```diff
@@ -128,6 +128,10 @@ DECLARE_INTERFACE_IID_(IBootstrapperEngine, IUnknown, "6480D616-27A0-44D7-905B-8
         __in DWORD dwWaitForInputIdleTimeout
         ) = 0;
 
+    STDMETHOD(SetUpdateSource)(
+        __in_z LPCWSTR wzUrl
+        ) = 0;
+
```
Comment (rseanhall) [inline src/balutil/inc/IBootstrapperEngine.h:134]: This isn't sufficient. You have to update IEngine and Engine as well so the BA can actually call it.

Comment (rseanhall) [reply src/balutil/inc/IBootstrapperEngine.h:134]: Sorry, I thought this was the managed `IBootstrapperEngine`. You'll have to update IBootstrapperEngine.cs plus IEngine.cs plus Engine.cs.

Comment (robmen) [reply src/balutil/inc/IBootstrapperEngine.h:134]: Yeah, I knew I was going to miss one of those connections.

File: src/WixToolset.Mba.Core/IBootstrapperEngine.cs (hunk 1 above)
Comment (robmen) [reply src/WixToolset.Mba.Core/IBootstrapperEngine.cs:506]: Yeah, I'm thinking about moving the `WixToolset.BoostrapperCore.Native` headers to this repo to make this the "Burn API" repo but I haven't done that yet so I still needed to update the `packages.config`. 😄

Comment (rseanhall) [reply src/WixToolset.Mba.Core/IBootstrapperEngine.cs:514]: You would make Burn repo depend on balutil repo? Weird. I would rather combine the repos.

Comment (robmen) [reply src/WixToolset.Mba.Core/IBootstrapperEngine.cs:514]: They should all be updated at the same time to have the same voice. My doc brain wasn't turned on while trying to make sure I got everything ported correctly. 😄

**Pull request merged**